clang/lib/CIR/CodeGen/CIRGenBuiltinX86.cpp

```diff
@@ -89,6 +89,14 @@ static mlir::Value getMaskVecValue(CIRGenFunction &cgf, const CallExpr *expr,
   }
   return maskVec;
 }
+static mlir::Value emitX86CompressExpand(CIRGenFunction &cgf, const CallExpr *expr,ArrayRef<mlir::Value> ops, bool IsCompress, const std::string &ID){
+  auto ResultTy = cast<cir::VectorType>(ops[1].getType());
+  mlir::Value MaskValue = getMaskVecValue(cgf, expr, ops[2],  cast<cir::VectorType>(ResultTy).getSize());
+  llvm::SmallVector<mlir::Value, 4> op{ops[0], ops[1], MaskValue};
+  
+  return emitIntrinsicCallOp(cgf,expr, ID, ResultTy, op);
```

Comment:
```suggestion
  return emitIntrinsicCallOp(cgf,expr, ID, ResultTy, { ops[0], ops[1], maskValue });
```
I don't think you need an explicit local to hold the array. Also, since you're passing the intrinsic name in as a string, the `isCompress` argument isn't needed.